### PR TITLE
Add azureResourceID and OMScloudID in the httpRequest headers

### PR DESCRIPTION
**Pull request by sugr4**
for all output plugins

@Microsoft/omsagent-devs

### CODE REVIEW

File: source/code/plugins/oms_common.rb [hunk 1]
```diff
@@ -674,9 +674,18 @@ def create_ods_http(ods_uri, proxy={})
       #   serializer: method. serializer of the record
       # returns:
       #   HTTPRequest. request to ODS
-      def create_ods_request(path, record, compress, extra_headers=nil, serializer=method(:parse_json_record_encoding))
+      def create_ods_request(path, omsadmin_conf_path, record, compress, extra_headers=nil, serializer=method(:parse_json_record_encoding))
```
Comment (robbiezhang) [inline source/code/plugins/oms_common.rb]: please use the oms_configuration file to load all configurations in a batch.

**Comment by robbiezhang** [comment]
please use the oms_configuration file to load all configurations in a batch. looking forward to another iteration.

File: source/code/plugins/oms_common.rb [hunk 2]
```diff
@@ -674,9 +674,18 @@ def create_ods_http(ods_uri, proxy={})
...
         headers = extra_headers.nil? ? {} : extra_headers
 
+        File.open(omsadmin_conf_path).each_line do |line|
```
Comment (varunkumta) [inline source/code/plugins/oms_common.rb]: we shouldn't read the conf file for every request .. instead have it static in memory and read the file only if it is not initialized (first time after service is started/restarted)

**Comment by sugr4** [comment]
Used oms_configuration.rb to load values for azureResourceID and omscloudID

File: source/code/plugins/oms_configuration.rb [hunk 3]
```diff
@@ -99,6 +101,15 @@ def load_configuration(conf_path, cert_path, key_path)
           Log.warn_once("Found more than one AGENT_GUID setting in #{conf_path}, will use the first one.")
         end
 
+        File.open(conf_path).each_line do |line|
```
Comment (robbiezhang) [inline source/code/plugins/oms_configuration.rb]: please move the section down after parsing the AgentId below.

**Comment by robbiezhang** [comment]
Signoff

**Pull request closed without merging**